modal/_container_entrypoint.py

```diff
@@ -579,13 +579,29 @@ def import_function(function_def: api_pb2.Function, ser_cls, ser_fun, ser_params
     # This is not in function_io_manager, so that any global scope code that runs during import
     # runs on the main thread.
     module: Optional[ModuleType] = None
+    cls: Optional[Type] = None
+    fun: Callable
     if ser_fun is not None:
         # This is a serialized function we already fetched from the server
         cls, fun = ser_cls, ser_fun
     else:
         # Load the module dynamically
         module = importlib.import_module(function_def.module_name)
-        cls, fun = load_function_from_module(module, function_def.function_name)
+        qual_name: str = function_def.function_name
+
+        if not is_global_function(qual_name):
+            raise LocalFunctionError("Attempted to load a function defined in a function scope")
+
+        parts = qual_name.split(".")
+        if len(parts) == 1:
+            cls = None
+            fun = getattr(module, qual_name)
+        elif len(parts) == 2:
+            cls_name, fun_name = parts
+            cls = getattr(module, cls_name)
+            fun = getattr(cls, fun_name)
```

Comment:
in particular this line can be rewritten quite a bit – `cls` should generally be a `Cls` object, so we can look up the method directly.

(this isn't always true in weird cases like if `@stub.cls` was not applied in global scope, to be strict)

Reply:
I thought `@stub.cls` not in global scope shouldn't work anyway because the function qualname will have 3 parts

Reply:
I was talking about the case where the class is _defined_ in global scope, but wrapped in local scope, i.e.

```python
import modal

stub = modal.Stub()

class Cls:
    @modal.method()
    def f(self):
        return 123


if __name__ == "__main__":
    Cls = stub.cls()(Cls)
    with stub.run():
        print(Cls().f.remote())
```

This actually fails right now but I don't think for reasons that have to do with qualnames.